### ISSUE TITLE
Unable to build singularity image

### ISSUE THREAD

**Issue by Elveera28** [VolundurH/wgs_fusion_pipeline#1]
Hello,

I'm Elveera and I'm very much interested in using this pipeline in validating the fusions in my WGS dataset.
but I'm not able to create the .sif image file from the .def file using commands provided 
my steps are as follows:
I've cloned this repository using git clone https://github.com/VolundurH/wgs_fusion_pipeline.git
I've created singularity env using the command :
conda create -n singularity -c conda-forge singularity=3.8.6 
and activated it using : conda activate singularity 
after this I'm running the command: 
singularity build --fakeroot singularity/fusion_pipeline.sif singularity/fusion_pipeline.def

this starts the build process and gives the following error and exits:
+ mamba install -c bioconda -c conda-forge -c anaconda -y samtools=1.13 bedtools=2.29.2 r-tidyverse=1.3.1 bioconductor-iranges=2.28.0 novoalign=3.09.00

Looking for: ['samtools=1.13', 'bedtools=2.29.2', 'r-tidyverse=1.3.1', 'bioconductor-iranges=2.28.0', 'novoalign=3.09.00']

[+] 0.0s
bioconda/linux-64 (check zst)-----------------------------------------------------------------------------------------------------------this goes on--------------------------------------------------------------------------------------------------------------------------------------

Pinned packages:
  - python 3.12.*

warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Could not solve for environment specs
The following packages are incompatible
├─ bioconductor-iranges 2.28.0**  is installable and it requires
│  └─ bioconductor-biocgenerics >=0.40.0,<0.41.0 , which requires
│     └─ r-base >=4.1,<4.2.0a0  with the potential options
│        ├─ r-base [4.1.0|4.1.1] would require
│        │  └─ icu >=68.1,<69.0a0 , which can be installed;
│        ├─ r-base [4.1.1|4.1.2|4.1.3] would require
│        │  └─ icu >=69.1,<70.0a0 , which can be installed;
│        ├─ r-base 4.1.3 would require
│        │  └─ icu >=70.1,<71.0a0 , which can be installed;
│        ├─ r-base 4.1.3 would require
│        │  └─ ncurses >=6.4,<7.0a0 , which can be installed;
│        ├─ r-base 4.1.3 would require
│        │  ├─ icu >=72.1,<73.0a0 , which conflicts with any installable versions previously reported;
│        │  └─ ncurses >=6.3,<7.0a0 , which can be installed;
│        └─ r-base 4.1.3 would require
│           ├─ icu >=72.1,<73.0a0 , which conflicts with any installable versions previously reported;
│           └─ ncurses >=6.4,<7.0a0 , which can be installed;
├─ mamba is installable with the potential options
│  ├─ mamba [1.5.7|1.5.8] would require
│  │  ├─ conda >=24,<25  with the potential options
│  │  │  ├─ conda [23.10.0|23.11.0|...|24.3.0] would require
│  │  │  │  └─ python >=3.10,<3.11.0a0 , which can be installed;
│  │  │  ├─ conda [23.10.0|23.11.0|...|24.3.0] would require
│  │  │  │  └─ python >=3.11,<3.12.0a0 , which can be installed;
│  │  │  ├─ conda [23.10.0|23.11.0|...|24.3.0] would require
│  │  │  │  └─ python >=3.8,<3.9.0a0 , which can be installed;
│  │  │  ├─ conda [23.10.0|23.11.0|...|24.3.0] would require
│  │  │  │  └─ python >=3.9,<3.10.0a0 , which can be installed;
│  │  │  ├─ conda [24.1.1|24.1.2|24.3.0] would require
│  │  │  │  └─ python >=3.12,<3.13.0a0 , which requires
│  │  │  │     └─ ncurses >=6.4,<7.0a0 , which can be installed;
│  │  │  └─ conda [24.1.0|24.1.1|24.1.2|24.3.0], which can be installed;
│  │  └─ libmambapy [1.5.7 py312hd9e9ff6_0|1.5.8 py312hd9e9ff6_0], which requires
│  │     └─ libmamba [1.5.2 had39da4_1|1.5.3 had39da4_1|...|1.5.8 had39da4_0], which requires
│  │        └─ libarchive >=3.7.2,<3.8.0a0  but there are no viable options
│  │           ├─ libarchive 3.7.2 would require
│  │           │  └─ libxml2 >=2.11.5,<3.0.0a0  but there are no viable options
│  │           │     ├─ libxml2 2.11.5 would require
│  │           │     │  └─ icu >=72.1,<73.0a0 , which conflicts with any installable versions previously reported;
│  │           │     └─ libxml2 [2.11.5|2.11.6|...|2.12.6] would require
│  │           │        └─ icu >=73.2,<74.0a0 , which conflicts with any installable versions previously reported;
│  │           └─ libarchive 3.7.2 would require
│  │              └─ libxml2 >=2.12.2,<3.0.0a0 , which cannot be installed (as previously explained);
│  ├─ mamba [0.0.10|0.0.7|0.0.8|0.0.9|0.1.0] would require
│  │  └─ python >=2.7,<2.8.0a0  with the potential options
│  │     ├─ python [2.7.12|2.7.13|...|2.7.18], which can be installed;
│  │     └─ python [2.7.15|3.6.7|3.7.1] would require
│  │        └─ openssl >=1.1.1,<1.1.2.0a0 , which does not exist (perhaps a missing channel);
│  ├─ mamba [0.0.10|0.0.7|...|0.9.2] would require
│  │  └─ python >=3.6,<3.7.0a0  with the potential options
│  │     ├─ python [3.6.0|3.6.1|...|3.6.9], which can be installed;
│  │     └─ python [2.7.15|3.6.7|3.7.1], which cannot be installed (as previously explained);
│  ├─ mamba [0.0.10|0.0.7|...|0.9.2] would require
│  │  └─ python >=3.7,<3.8.0a0  with the potential options
│  │     ├─ python [3.7.0|3.7.1|...|3.7.9], which can be installed;
│  │     └─ python [2.7.15|3.6.7|3.7.1], which cannot be installed (as previously explained);
│  ├─ mamba [0.1.2|0.10.0|...|1.5.8] would require
│  │  └─ python >=3.8,<3.9.0a0 , which can be installed;
│  ├─ mamba [0.10.0|0.11.1|...|1.5.8] would require
│  │  └─ python >=3.9,<3.10.0a0 , which can be installed;
│  ├─ mamba [0.18.1|0.18.2|...|1.5.8] would require
│  │  └─ python >=3.10,<3.11.0a0 , which can be installed;
│  ├─ mamba [1.0.0|1.1.0|...|1.5.8] would require
│  │  └─ python >=3.11,<3.12.0a0 , which can be installed;
│  └─ mamba [1.5.2|1.5.3|1.5.4|1.5.5|1.5.6] would require
│     ├─ conda [>=23.11,<23.12 |>=23.9,<23.11.0 ] with the potential options
│     │  ├─ conda [23.10.0|23.11.0|...|24.3.0], which can be installed (as previously explained);
│     │  ├─ conda [23.10.0|23.11.0|...|24.3.0], which can be installed (as previously explained);
│     │  ├─ conda [23.10.0|23.9.0] would require
│     │  │  └─ python >=3.12,<3.13.0a0 , which can be installed (as previously explained);
│     │  ├─ conda [23.10.0|23.11.0|...|24.3.0], which can be installed (as previously explained);
│     │  ├─ conda [23.10.0|23.11.0|...|24.3.0], which can be installed (as previously explained);
│     │  └─ conda 23.11.0 would require
│     │     └─ python >=3.12,<3.13.0a0 , which can be installed (as previously explained);
│     └─ libmambapy [1.5.2 py312hd9e9ff6_1|1.5.3 py312hd9e9ff6_1|...|1.5.6 py312hd9e9ff6_0], which cannot be installed (as previously explained);
├─ python is installable with the potential options
│  ├─ python [3.12.0|3.12.1|3.12.2], which can be installed (as previously explained);
│  ├─ python [1.0.1|1.2|...|3.6.0rc1], which can be installed;
│  ├─ python [2.7.12|2.7.13|...|2.7.18], which can be installed;
│  ├─ python [3.10.0|3.10.1|...|3.10.9], which can be installed;
│  ├─ python [3.11.0|3.11.1|...|3.11.8], which can be installed;
│  ├─ python [3.6.0|3.6.1|...|3.6.9], which can be installed;
│  ├─ python [3.7.0|3.7.1|...|3.7.9], which can be installed;
│  ├─ python [3.8.0|3.8.1|...|3.8.8], which can be installed;
│  ├─ python [3.9.0|3.9.1|...|3.9.9], which can be installed;
│  ├─ python 3.12.0rc3 would require
│  │  └─ _python_rc, which does not exist (perhaps a missing channel);
│  └─ python [2.7.15|3.6.7|3.7.1], which cannot be installed (as previously explained);
└─ samtools 1.13**  is not installable because it requires
   └─ ncurses >=6.2,<6.3.0a0 , which conflicts with any installable versions previously reported.
FATAL:   While performing build: while running engine: exit status 1

It would be great if you could help me resolve this error.
thank you.

**Comment by VolundurH** [comment]
Hi Elveera, thanks for taking an interest in the pipeline!

I have pushed a change to the .def file that should fix your issue
Please let me know if you have any further issues

Best,
Völundur

**Comment by Elveera28** [comment]
Hello,
Thank you so much for your quick response and the fix! I was able to build the image file without error. 

However I've further encountered the following error while running the Rscript using the following command:
singularity exec --bind /disk/projects/data:/data singularity/fusion_pipeline.sif Rscript scripts/fusion_validation_pipeline.R

The error I got is as follows:
During startup - Warning messages:
1: Setting LC_CTYPE failed, using "C"
2: Setting LC_COLLATE failed, using "C"
3: Setting LC_TIME failed, using "C"
4: Setting LC_MESSAGES failed, using "C"
5: Setting LC_MONETARY failed, using "C"
6: Setting LC_PAPER failed, using "C"
7: Setting LC_MEASUREMENT failed, using "C"
-- Attaching packages -------------------------------------------------------------------------------------- tidyverse 1.3.2 --
v ggplot2 3.4.2     v purrr   1.0.1
v tibble  3.2.1     v dplyr   1.1.2
v tidyr   1.3.0     v stringr 1.5.0
v readr   2.1.4     v forcats 1.0.0
-- Conflicts ----------------------------------------------------------------------------------------- tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
Warning messages:
1: package 'tidyverse' was built under R version 4.1.3
2: Failed to locate timezone database
3: package 'ggplot2' was built under R version 4.1.3
4: package 'tibble' was built under R version 4.1.3
5: package 'tidyr' was built under R version 4.1.3
6: package 'readr' was built under R version 4.1.3
7: package 'purrr' was built under R version 4.1.3
8: package 'dplyr' was built under R version 4.1.3
9: package 'stringr' was built under R version 4.1.3
10: package 'forcats' was built under R version 4.1.3
Error in library(optparse) : there is no package called 'optparse'
Execution halted

Thanking you in advance.
